homeassistant/components/bluetooth/match.py

```diff
@@ -80,12 +80,15 @@ def seen_all_fields(
         or not previous_match.service_data.issuperset(advertisement_data.service_data)
     ):
         return False
-    if advertisement_data.service_uuids and (
-        not previous_match.service_uuids
-        or not previous_match.service_uuids.issuperset(advertisement_data.service_uuids)
-    ):
-        return False
-    return True
+    return not (
+        advertisement_data.service_uuids
+        and (
+            not previous_match.service_uuids
+            or not previous_match.service_uuids.issuperset(
+                advertisement_data.service_uuids
+            )
+        )
+    )
```

Comment:
I would use a `noqa` here. We have a specific list of conditionals for which the return is False and as last statement a return True.